src/main/java/org/apache/maven/model/jdom/util/JDomUtils.java

```diff
@@ -50,7 +51,7 @@ private JDomUtils() {
    * @param root    the root element.
    */
   public static void addElement(Element element, Element root) {
```

Comment:
Maybe you should update the JavaDoc: `Add as last element` or `Append new element`

Reply:
It isn't always added as last element. I'd rather replicate more detailed information from the Javadoc of the `#inserNewElement` method

Reply:
The JavaDocs says: 
```
/**
   * Adds an element as new child to the given root element. The position where the element is inserted is calculated
   * using the element order that is defined in the {@link JDomCfg} (see {@link JDomCfg#getElementOrder(String)}). When
   * no order is defined for the element, the new element is append as last element (before the closing tag of the
   * root element). In the root element, the new element is always prepended by a text element containing a linebreak
   * followed by the indentation characters. The indentation of the new element nodes will be reset to the indentation
   * that is (tried to be) detected from the root element (see {@link #detectIndentation(Element)}).
   */
```
The implementation is `getLastElementIndex(root) + 1`: Why it isn't always added as last element then?

Reply:
Maybe we should distinguish between element index and content index. The last child (content) is always the `org.jdom2.Text` containing the line break and indentation before the end tag of the enclosing element.